prism/prism.py

```diff
@@ -98,364 +393,1074 @@ def create_bearer_token(self):
+            limit=None, offset=None,
+            type_="summary",
+            search=False):
+        """Obtain details for all tables or a given table(s).
+
+        Notes
+        -----
+            This method never fails and always returns a valid Dict.
 
         Parameters
         ----------
         table_name : str
-            The table name. The name must be unique and conform to the name
-            validation rules.
-
-        schema : list
-            A list of dictionaries containing the schema
+            The name of the table to obtain details about. If the default value
+            of None is specified.
+        table_id : str
+            The ID of a table to obtain details about.  When specified, all tables
+            are searched for the matching id.
+        limit : int
+            The maximum number of tables to be queried, if None all tables are returned.
+        offset: int
+            The offset from zero of tables to return.
+        type_ : str
+            Level of detail to return.
+        search : bool
+            Enable contains searching for table names and display names.
 
         Returns
         -------
-        If the request is successful, a dictionary containing information about
-        the new table is returned.
-
+        dict
+            For an ID query, return the table information as a dict.  For any other
+            table list query, return a total attribute of the number of tables found and data
+            attribute containing the list tables.
         """
-        url = self.prism_endpoint + "/datasets"
+        operation = "/tables"
+
+        if type_ is None or type_.lower() not in ["full", "summary", "permissions"]:
+            logger.warning("Invalid output type for tables list operation - defaulting to summary.")
+            output_type = "summary"
+        else:
+            output_type = type_.lower()
 
-        headers = {
-            "Authorization": "Bearer " + self.bearer_token,
-            "Content-Type": "application/json",
+        # If we got a WID, then do a direct query by ID - no paging or searching required.
+        if table_id is not None:
+            operation = f"{operation}/{table_id}?format={output_type}"
+            logger.debug(f"get: {operation}")
+            url = self.prism_endpoint + operation
+
+            response = self.http_get(url)
+
+            if response.status_code == 200:
+                return response.json()
+            else:
+                return None
+
+        # We are doing a query by attributes other than ID.
+        logger.debug(f"get: {operation}")
+        url = self.prism_endpoint + operation
+
+        # Always return a valid JSON object of results regardless of
+        # errors or API responses.  THIS METHOD NEVER FAILS.
+        return_tables = {"total": 0, "data": []}
+
+        # Start setting up the API call parameters.
+        params = {
+            'limit': limit if limit is not None else 100,
```

Comment:
For all of these parameters, why not just set the default values when defining the function?

Reply:
I'm using None as a cheap way to tell if the user wants everything, i.e., no limit / all items.  There is a hole in my logic I need to think about - standby.

Reply:
So, when the caller does NOT pass a values, i.e., None, this means get all.  I thought this would be better than using, say "-1" or "999" as a limit value - especially since these are invalid on the real API call (don't want to confuse the user).  When the user does a non-table search call and does provide a limit, this routine works exactly like standard API call.  **The value-add here is I don't force them to accept a limit.**

If the user does a table search, limits are ignored because I have to pass all the pages of tables to find matches.  Also, if I tried to apply limits to a search for table names then I'm just replicating the API limit/offset functionality.

**Question**: As I think about this, this functionality may be better placed in the cli.py + tables_commands.py and not in prism.py class, but it seem generically useful.  Any thoughts on moving the logic out of prism.py?

Reply:
I see.  I think it's fine to keep it here.